coils_tet_mesh.py

```diff
@@ -0,0 +1,277 @@
+            csv_path (str): absolute path to a .csv file that will store the top 
+                quality metrics data for each coil.
+        """
+
+    # GEOMETRY AND MESH CREATION
+
+    # Start Cubit - this step is key
+    cubit.init(['cubit', '-nojournal'])
+
+    # Import coils
+    cubit.cmd(f'import step "{coils_path}" heal')
+    cubit.cmd('volume  size auto factor 5')
+
+    # Split log path
+    log_split = log_general_path.split('.')
+    log_prefix = log_split[0]
+    log_ext = log_split[1]
+
+    # Define mesh quality metrics
+    quality_metric_list = ['shape', 'aspect ratio', 'condition no.',
+                           'distortion', 'element volume','equiangle skew', 'equivolume skew',
+                           'inradius', 'jacobian', 'normalized inradius', 'relative size',
+                           'scaled jacobian', 'shape and size']
+    
+    # Create an empty Pandas data frame to store the top metric results
+    top_results_cols = ['Function Name', 'Average','Standard Deviation',
+                                        'Minimum', 'Maximum']
+    top_results = pd.DataFrame(columns = top_results_cols)
+
+    # For-loop iterates over all of the coils included in the coils.step file
+    for coil_num in np.arange(1,total_coil_count + 1):
+        
+        # Create the mesh
+        cubit.cmd(f'volume {coil_num} scheme tetmesh')
+        cubit.cmd(f'mesh volume {coil_num}')
+
+        # Set log file path and initiate logging
+        log_path = f"{log_prefix}_{coil_num}.{log_ext}"
+        cubit.cmd(f'logging on file "{log_path}"')
+
+        # Run mesh quality metrics for each relevant metric
+        for metric in quality_metric_list:
+            qual_command = f'quality volume {coil_num} {metric} global draw mesh list'
+            mesh_analytics = cubit.cmd(qual_command)
+
+
+        # MESH QUALITY ANALYSIS
+
+        # Read the log file
+        with open(log_path, 'r') as file:
+            text = file.read()
+
+        # Extract the data from the log file using the extract_tabular_data function
+        function_names, averages, std_devs, minima, maxima = extract_tabular_data(text)
+
+        # Separate multi-word function names to be able to match-case with Cubit names
+        function_names = split_function_names(function_names)
+
+        # Compile the extracted tabular data into a Pandas data frame
+        quality_metrics_df = pd.DataFrame({'Function Name': function_names,
+                                           'Average': averages,
+                                           'Standard Deviation': std_devs,
+                                           'Minimum': minima,
+                                           'Maximum': maxima})
+
+        # Remove rows where the standard deviation equals 0
+        # Found this for a few cases and it appeared too implausible to keep
+        quality_metrics_df = quality_metrics_df.loc[quality_metrics_df['Standard Deviation'] != 0]
```

Comment:
Separation of concerns & cognitive load:
Make this a method that builds & cleans the dataframe